mirgecom/__init__.py

```diff
@@ -25,26 +25,23 @@
 import sys
 from warnings import warn
 
-try:
-    from mpi4py import MPI
+from mpi4py import MPI
 
-    # This code avoids slow startups due to file system locking when running with
-    # large numbers of ranks. See https://github.com/illinois-ceesd/planning/issues/27
-    # for details.
+# This code avoids slow startups due to file system locking when running with
+# large numbers of ranks. See https://github.com/illinois-ceesd/planning/issues/27
+# for details.
 
-    size = MPI.COMM_WORLD.Get_size()
+size = MPI.COMM_WORLD.Get_size()
 
-    if size >= 128:
-        for mod in ["pyopencl", "pytools", "loopy"]:
-            if mod in sys.modules:
-                warn("{} already loaded, not adjusting XDG_CACHE_HOME.".format(mod))
+if size >= 128:
+    for mod in ["pyopencl", "pytools", "loopy"]:
+        if mod in sys.modules:
+            warn("{} already loaded, not adjusting XDG_CACHE_HOME.".format(mod))
 
-        if "XDG_CACHE_HOME" in os.environ:
-            warn("XDG_CACHE_HOME already set, not adjusting it.")
-        else:
-            rank = MPI.COMM_WORLD.Get_rank()
-            username = getpass.getuser()
+    if "XDG_CACHE_HOME" in os.environ:
+        warn("XDG_CACHE_HOME already set, not adjusting it.")
+    else:
+        rank = MPI.COMM_WORLD.Get_rank()
+        username = getpass.getuser()
 
-            os.environ["XDG_CACHE_HOME"] = "/tmp/{}/xdg-cache-r{}".format(username, rank)
-except:
-    pass
+        os.environ["XDG_CACHE_HOME"] = "/tmp/{}/xdg-cache-r{}".format(username, rank)
```

Comment:
We should conceivably create this directory.

Reply:
From my tests, these directories seem to be created automatically. Is it necessary to create them ourselves?

Reply:
Don't know. Check the XDG spec?

Reply:
XDG spec is unclear, creating the dir just to be safe.